### DISCUSSION TITLE
LSEP characters pasted and invisible in text-only fields

### DISCUSSION THREAD

**Discussion by juresrpcic** [craftcms/cms#8493] [Ideas] [answered]
### Description

I'm having issues with seeing LSEP characters in the frontend coming from text-only fields (both inside matrix blocks or directly attached to entry). This is how it looks on the site:

![lsep-frontend](https://user-images.githubusercontent.com/2803433/35638008-339081c8-06b6-11e8-9329-adaa9e84f288.jpg)

And this is the text block including these characters in the Control Panel:

![lsep-text-field](https://user-images.githubusercontent.com/2803433/35638019-3fed1418-06b6-11e8-85b0-79b85d598424.jpg)

**There is no visual clue that this character is hiding there.** If you put the cursor where you know it is, you can press delete/backspace two times and you're deleting the characters but nothing is visible - the third press will start deleting letters.

**Ideally, the text-only field should clean this character on paste or at least show there's something there.** They don't show in the frontend in all cases (visible in Chrome on Win 10, not visible in Chrome on OSX, also depends on fonts being used etc.).

I tried instructing the clients to paste text to a text-only program (eg. TextEdit on text mode) but as they are on OSX and don't even see the offending characters in frontend, they can't really see or fix this.

### Steps to reproduce

1. Copy text from PowerPoint that has a line break
2. Paste into a plain text field with multiline enabled
3. See invisible LSEP characters in the frontend, but not in the field when editing

### Additional info

- Craft version: 3.0.0-RC8
- PHP version: PHP 7.1.10 (cli) (built: Oct 4 2017 08:54:01)
- Database driver & version: MySQL 5.7.19

**Comment by brandonkelly** [comment]
This also likely affects Craft 2, across all `text` inputs and `textarea`s.

**Comment by ponomarchukov** [comment]
That is problem in database,  LSEP character (U+2028). Did you find solution how to remove?

**Comment by nodoxi** [comment]
I agree that textfields should clean this character on saving.

**Reply by brandonkelly** [reply]
What version of Craft are you running? Per https://github.com/craftcms/cms/discussions/8493#discussioncomment-1843996, line separator characters should be getting converted to newlines as of 3.0.0-RC17.1.

**Reply by nodoxi** [reply]
Hm. I can't reproduce this bug. But sometimes it appears in single-line textfields.

**Reply by brandonkelly** [reply]
Perhaps they are using a different field type than Plain Text?

**Reply by kevinmu17** [reply]
Redactor has a HTML cleaner, this symbol should be part of the cleaning process imo.

**Comment by brandonkelly** [comment, marked as answer]
Just tested this locally and line separator characters (`U+2028`) are getting converted to normal newline characters here, as they are covered by `\R` in unicode:

https://github.com/craftcms/cms/blob/97c8598b3c71a1b88e6bbe4ee35c3720e5fa5d7e/src/fields/PlainText.php#L213

That line has been in place since 3.0.0-RC17.1 – a bit after this suggestion was posted. So seems like this was resolved unintentionally.
